### PR TITLE
sys-devel/clang: fix path for clang-tblgen to allow cross-compile with cross-emerge, part of #667094 [do not merge]

### PR DESCRIPTION
**Pull request by stefson**
@mgorny 

emerge-armv6j-unknown-linux-gnueabihf -av clang:6 must fail, because clang-tblgen would be installed into `/usr/lib/llvm/6/bin/` and not into `/usr/bin` as it propably was with < sys-devel/clang-4 

- [ ] clang-tblgen is compiled on the hosts side, but not installed during src_install. fix that with fix-tblgen.patch for all >=clang-4 

- [ ] clang-tblgen cannot be found when tc-is-cross-compiler, as it assumes it to be in `/usr/bin` and not /`usr/lib/llvm/6/bin/`

asking for feedback mostly, cc-ing @chewi as mgorny told me you're doing a lot of cross compiling.

### CODE REVIEW

**Comment by stefson** [comment]
@alonbl - you commited a lot of cross compile related things yesterday. Have you any experience with cross compile of clang?

**Comment by alonbl** [comment]
On Sat, Oct 6, 2018 at 9:14 AM stefson <notifications@github.com> wrote:
>
> @alonbl - you commited a lot of cross compile related things yesterday. Have you any experience with cross compile of clang?
>

No, I use gcc.

**Comment by mgorny** [comment]
Let's start with updating LLVM for desired state, then update clang to match.

**Comment by ConiKost** [comment]
Is this still needed at all?

**Comment by stefson** [comment]
yes, this still is a problem and not solved in tree. for more information: https://bugs.gentoo.org/731264

**Comment by jsmolic** [comment]
I'm also interested in this. @stefson Do you want to rebase the patches?

**Comment by chewi** [comment]
Sorry my stance on this still hasn't changed so I would not accept it. Please build clang-tblgen for the build system while building clang for the host system. If that turns out to be infeasible then I may reconsider but I know that CMake does at least have functionality for this situation.

**Comment by stefson** [comment]
I can only refere you back to this solution, it works: https://bugs.gentoo.org/731264#c5

I have no idea why its always refused to allow installation of clang-tblgen for the host, so just build your own copy of it via the info from the bug report linked above.

File: sys-devel/clang/clang-4.0.1.ebuild [hunk 1]
```diff
@@ -98,6 +98,10 @@ src_prepare() {
 	# (backport, temporary reverted upstream because of FreeBSD issues)
 	eapply "${FILESDIR}"/4.0.1/0001-Frontend-Correct-values-of-ATOMIC_-_LOCK_FREE-to-mat.patch
 
+	# add tblgen for cross compile
+	# https://bugs.gentoo.org/667094
+	"${FILESDIR}"/fix-tblgen.patch
```
Comment (scardracs) [inline sys-devel/clang/clang-4.0.1.ebuild:103]: Except the fact this is a really old version of clang, didn't you miss an eapply?

**Comment by scardracs** [comment]
Ping: I think that can be closed, right?

**Comment by stefson** [comment]
Yes, the relevant patches have been merged and the ebuilds are fixed as well.

**Comment by chewi** [comment]
For the record, I did end up relying on the host's Clang installation, but I was at least able to do it in a supported way. I realised that building a host Clang during the target build was infeasible, because it would probably have meant building LLVM too. The dependency is still not enforced by Portage, but we're considering doing something about that.

**Pull request closed without merging**